ui/src/components/Comment/index.tsx

```diff
@@ -92,6 +92,9 @@ const Comment = ({ objectId, mode, commentId }) => {
     if (!data) {
       return;
     }
+    if (data.count <= 3) {
+      data.list.sort((a, b) => a.created_at - b.created_at);
+    }
     if (pageIndex === 1 || pageIndex === 0) {
```

Comment:
Sorry, I missed some conditions yesterday.  `data.count` should be the total number of comments. Here you only need to sort the data loaded for the first time and with the same number of votes by time, so `data.list.length` and `vote_Count` should be used for judgment, and it should only be processed during the first load. So it should be like this

```javascript
  if (pageIndex === 1 || pageIndex === 0) {
      if (data.list.length <= 3) {
        data.list.sort((a, b) => {
          if (a.vote_count === b.vote_count) {
            return a.created_at - b.created_at;
          }
          return 1;
        });
      }
      setComments(data?.list);
    } else {
      setComments([...comments, ...data.list]);
    }
```

Reply:
![image](https://github.com/apache/incubator-answer/assets/50784434/f461d4d4-07f1-4c20-9803-5ff2bc3bdbe0)
Okay, there might be some confusion in the needs. I want to clarify the needs with you.

In my opinion, the main condition is to determine whether the total number of comments is greater than 3, and the comments less than or equal to 3 are displayed in order of time from old to new. For comments larger than 3, they will be displayed in two sortings initially and after expansion.

If the main condition is the number of comments obtained, then sorting from old to new should only be effective for cases less than 3. But the number of votes for this case is not a sorting condition.

In addition, the default sorting for "page 0" is votes from high to low and then time from old to new. So no need to reorder.

Reply:
>  the main condition is to determine whether the total number of comments is greater than 3.

Your understanding is right.

>  In addition, the default sorting for "page 0" is votes from high to low and then time from old to new. So no need to reorder.

To be exact, the default sorting should not be based on "page 0", but you sort based on whether the total number is greater than 3, because "page 0" is a condition used by the front end to distinguish whether to load more,when using the useQueryComments` method, the parameter passed is page = 0, and the actual call passed is page=1.

The code is correct now！

Finally, thank you very much for your carefulness and contribution！